packages/@uppy/companion/src/server/provider/Provider.js

```diff
@@ -4,11 +4,13 @@
 class Provider {
   /**
    *
-   * @param {object} options
+   * @param {{providerName: string, allowLocalUrls: boolean, providerOptions: object}} options
    */
-  constructor (options) { // eslint-disable-line no-unused-vars
+  constructor ({ allowLocalUrls, providerOptions }) {
     // Some providers might need cookie auth for the thumbnails fetched via companion
     this.needsCookieAuth = false
+    this.allowLocalUrls = allowLocalUrls
+    this.providerOptions = providerOptions
```

Comment:
Do these changes belong in this PR?

Reply:
`providerOptions` probably is not necessary at this point
`allowLocalUrls` is injected to the provider so we know what to pass to `validateUrl`, so I'd say: yes, this belongs into this PR

Reply:
but isn't this PR about simply making the helpers validateURL and getProtectedHttpAgent plublically exposed?

Reply:
Well, yes, but I need to be able to use it? Without introducing my own env var handling in my provider?

Reply:
Removed the `providerOptions` (to send them in another PR at some point) but I stick to it: it's a good idea and it belongs into this PR to make the global `allowLocalUrls` option available to the provider.